cocotb/scheduler.py

```diff
@@ -71,6 +72,14 @@
                            create_error, ExternalException)
 from cocotb.utils import nullcontext
 
+# On python 3.7 onwards, `dict` is guaranteed to preserve insertion order.
+# Since `OrderedDict` is a little slower that `dict`, we prefer the latter
+# when possible.
+if sys.version_info[:2] >= (3, 7):
+    _ordered_dict = dict
+else:
+    _ordered_dict = collections.OrderedDict
+
```

Comment:
It's surprising to me that Python wouldn't do that optimization by itself.

Reply:
OrderedDict has extra methods (that we don't use anyway), so I don't think it can